src/Cli/Microsoft.TemplateEngine.Cli/Commands/SharedOptions.cs

```diff
@@ -11,7 +12,23 @@ public static class SharedOptions
         {
             Description = SymbolStrings.Option_Output,
             Required = false,
-            Arity = new ArgumentArity(1, 1)
+            Arity = new ArgumentArity(1, 1),
+            CustomParser = (ArgumentResult argumentResult) =>
+            {
+                var input = argumentResult.Tokens[0]?.Value;
+                if (string.IsNullOrWhiteSpace(input))
+                {
+                    return null;
+                }
+                else if (Path.EndsInDirectorySeparator(input))
+                {
```

Comment:
Could simplify the else with something like

`return new FileInfo(input.TrimEnd(Path.DirectorySeparateChar)+Path.DirectorySeparateChar);`